pkgs/development/libraries/qt-6/patches/qtdeclarative-invalidate-caches-from-mismatched-store-paths.patch

```diff
@@ -0,0 +1,112 @@
+diff --git a/src/qml/jsruntime/qv4executablecompilationunit.cpp b/src/qml/jsruntime/qv4executablecompilationunit.cpp
+index 2ab38b70d9..8634b07e41 100644
+--- a/src/qml/jsruntime/qv4executablecompilationunit.cpp
++++ b/src/qml/jsruntime/qv4executablecompilationunit.cpp
+@@ -33,6 +33,8 @@
+ #include <QtCore/qcryptographichash.h>
+ #include <QtCore/QScopedValueRollback>
+ 
++#include <QtCore/qcoreapplication.h>
++
+ static_assert(QV4::CompiledData::QmlCompileHashSpace > QML_COMPILE_HASH_LENGTH);
+ 
+ #if defined(QML_COMPILE_HASH) && defined(QML_COMPILE_HASH_LENGTH) && QML_COMPILE_HASH_LENGTH > 0
+@@ -44,6 +46,36 @@ __attribute__((section(".qml_compile_hash")))
+ const char qml_compile_hash[QV4::CompiledData::QmlCompileHashSpace] = QML_COMPILE_HASH;
+ static_assert(sizeof(QV4::CompiledData::Unit::libraryVersionHash) > QML_COMPILE_HASH_LENGTH,
+               "Compile hash length exceeds reserved size in data structure. Please adjust and bump the format version");
++
++bool nix__isNixApplication() {
++  static const bool value = QCoreApplication::applicationFilePath().startsWith(QStringLiteral("/nix/store/"));
++  return value;
++}
++
++static_assert(sizeof(QV4::CompiledData::Unit::libraryVersionHash) >
++                  /*sha1*/ 20 + /*NIX:*/ 4,
++              "Nix compile hash length exceeds the reserved space in data "
++              "structure. Please review the patch.");
++
++const QByteArray &nix__applicationHash() {
++  static const QByteArray value = [](){
++        QCryptographicHash applicationHash(QCryptographicHash::Sha1);
++        applicationHash.addData(QByteArrayView(qml_compile_hash, QML_COMPILE_HASH_LENGTH));
++
++        // We only care about the package, not the specific file path.
++        auto view = QCoreApplication::applicationFilePath().sliced(11); // len("/nix/store/")
++        auto pkgEndIdx = view.indexOf(QStringLiteral("/"));
++        if (pkgEndIdx != -1) view = view.sliced(0, pkgEndIdx);
++
++        QCryptographicHash::hashLength(QCryptographicHash::Sha1);
```

Comment:
Unused? Unless I'm missing something

Reply:
yeah, was going to use that instead of hardcoding 20 but it wasn't constexpr and I forgot to remove it